### ISSUE TITLE
view support

### ISSUE THREAD

**Issue by mariusa** [tomyl/pg-dump-upsert#6]
Postgres' pg_dump can't dump data from a view.
Would it be possible to have this tool support views?
Thanks for considering.

**Comment by tomyl** [comment]
This already works. Have you encountered any problem with views?

```
$ cat example.sql                                      
create table employee (
    id integer primary key,
    created_at timestamptz not null default current_timestamp,
    name text not null,
    salary integer not null
);

insert into employee (id, name, salary) values (1, 'Alice', 123456);
insert into employee (id, name, salary) values (2, 'Bob', 90000);

create view test as select * from employee where salary >= 100000;

$ psql -h localhost -p 5432 -U postgres -f example.sql
Password for user postgres: 
Expanded display is used automatically.
Null display is "Ø".
CREATE TABLE
INSERT 0 1
INSERT 0 1
CREATE VIEW
$ ./pg-dump-upsert -dsn "postgresql://postgres:secret@localhost:5432/postgres?sslmode=disable" -table test -conflict-column id
INSERT INTO test (salary, name, created_at, id) VALUES (123456, 'Alice', '2021-11-25 21:26:07.188179+00', 1) ON CONFLICT (id) DO UPDATE SET salary=EXCLUDED.salary, name=EXCLUDED.name, created_at=EXCLUDED.created_at;
```

**Comment by mariusa** [comment]
Ah, this is good news! I've searched the repo for 'view' and got no results. Thanks for the fast reply and reproducible code.

There are 2 issues:
1. "camelCase" columns don't work. eg
```
create table employee (
    id integer primary key,
    "createdAt" TIMESTAMPTZ DEFAULT now(),
    created_at timestamptz not null default current_timestamp,
    name text not null,
    salary integer not null
);
```
gives `pq: column "createdat" does not exist`

2. (more a feature request than bug) 
GENERATED STORED columns aren't recognized, they should be exluded. `\copy` does exclude them from output
https://www.postgresql.org/docs/14/sql-copy.html
```
create table employee2 (
    id integer primary key,
    created_at timestamptz not null default current_timestamp,
    name text not null,
    salary integer not null,
    search tsvector GENERATED ALWAYS AS (to_tsvector('english', coalesce(name, '') )) STORED
);

create view test2 as select * from employee2;
./pg-dump-upsert ... -table test2 -conflict-column id
panic: don't know how to bind column search of type tsvector
```

**Comment by tomyl** [comment]
> 1. "camelCase" columns don't work. eg

Hopefully fixed in 3c8bd2fbe5fd7ce04cfc6e07b49e16727bfd850c.

> 
> 2. (more a feature request than bug)
>    GENERATED STORED columns aren't recognized, they should be exluded. `\copy` does exclude them from output

Ah makes sense. I supposed this is `is_generated` in `information_schema.columns`.

> panic: don't know how to bind column search of type tsvector

This type wasn't supported yet. Hopefully fixed in 25a08e04b11efbdc7ab45cdad6fb5060e85d440a.

**Comment by mariusa** [comment]
Thank you so much, Tommy!
